sdk/eventgrid/Azure.Messaging.EventGrid/src/MigrationGuide.md

```diff
@@ -0,0 +1,195 @@
+EventGridClient client = new EventGridClient(topicCredentials);
+```
+
+In the `Azure.Messaging.EventGrid` library, the client used to interact with the service was renamed as the `EventGridPublisherClient` to highlight the fact that the client is only used to publish events and does not directly consume events from the service. The constructor takes the topic endpoint in addition to a credential parameter since a single client will only ever be able to publish to a single topic.
+```C# Snippet:CreateClientWithOptions
+EventGridPublisherClient client = new EventGridPublisherClient(
+    new Uri(topicEndpoint),
+    new AzureKeyCredential(topicAccessKey));
+```
+
+### Publishing events to a topic
+
+In the `Microsoft.Azure.EventGrid` library, events were published as shown:
+```C#
+string topicEndpoint = "https://<topic-name>.<region>-1.eventgrid.azure.net/api/events";
+string topicHostname = new Uri(topicEndpoint).Host;
+
+List<EventGridEvent> eventsList = new List<EventGridEvent>();
+    for (int i = 0; i < 1; i++)
+    {
+        eventsList.Add(new EventGridEvent()
+        {
+            Id = Guid.NewGuid().ToString(),
+            EventType = "Contoso.Items.ItemReceivedEvent",
+            Data = new ContosoItemReceivedEventData()
+            {
+                ItemUri = "ContosoSuperItemUri"
+            },
+
+            EventTime = DateTime.Now,
+            Subject = "Door1",
+            DataVersion = "2.0"
+        });
+    }
+
+await client.PublishEventsAsync(topicHostname, eventsList);
+```
+
+In `Azure.Messaging.EventGrid`, when publishing events it is no longer necessary to include the topic host name, as this is instead passed as part of the `Uri` that is provided to the `EventGridPublisherClient` constructor. The `EventGridEvent` type now has a constructor that includes parameters for all required fields of the event grid schema:
+```C# Snippet:SendEGEventsToTopic
+// Add EventGridEvents to a list to publish to the topic
+List<EventGridEvent> eventsList = new List<EventGridEvent>
+{
+    new EventGridEvent(
+        "ExampleEventSubject",
+        "Example.EventType",
+        "1.0",
+        "This is the data for the first event"),
+   new EventGridEvent(
+        "ExampleEventSubject",
+        "Example.EventType",
+        "1.0",
+        "This is the data for the second event")
+};
+
+// Send the events
+await client.SendEventsAsync(eventsList);
+```
+### Deserializing events and data
+
+In `Microsoft.Azure.EventGrid`, there was the `EventGridSubscriber` type that was used to deserialize events:
+```C#
+string requestContent = await req.Content.ReadAsStringAsync();
+
+EventGridSubscriber eventGridSubscriber = new EventGridSubscriber();
+
+// Optionally add one or more custom event type mappings
+eventGridSubscriber.AddOrUpdateCustomEventMapping("Contoso.Items.ItemReceived", typeof(ContosoItemReceivedEventData));
+
+var events = eventGridSubscriber.DeserializeEventGridEvents(requestContent);            
```

Comment:
Should we use an explicit type here for consistency?

Reply:
@pakrym suggested we just use a stream instead of HttpContent.

Reply:
I meant to add this comment to your other comment.. Yes, I can add an explicit type here.